src/apps/filters/h3.c

```diff
@@ -1189,6 +1189,433 @@ SUBCOMMAND(uncompactCells,
     return E_SUCCESS;
 }
 
+/// Region subcommands
+
+H3Error polygonStringToGeoPolygon(FILE *fp, char *polygonString,
+                                  GeoPolygon *out) {
+    // There are two kinds of valid input, an array of arrays of lat, lng values
+    // defining a singular polygon loop, or an array of array of arrays of lat,
+    // lng values defining a polygon and zero or more holes. Which kind of input
+    // this is can be determined by the `[` depth reached before the first
+    // floating point number is found. This means either 2 or 3 `[`s at the
+    // beginning are valid, and nothing more than that.
+    int8_t maxDepth = 0;
+    int8_t curDepth = 0;
+    int64_t numVerts = 6;
+    int64_t curVert = 0;
+    int64_t curLoop = 0;
+    LatLng *verts = calloc(numVerts, sizeof(LatLng));
+    int strPos = 0;
+    while (polygonString[strPos] != 0) {
+        // Load more of the file if we've hit our buffer limit
+        if (strPos >= 1500 && fp != 0) {
+            int res = fread(polygonString, 1, 1500, fp);
+            strPos = 0;
+            // If we didn't get any data from the file, we're done.
+            if (res == 0) {
+                break;
+            }
+        }
+        // Try to match `[` first
+        if (polygonString[strPos] == '[') {
+            curDepth++;
```

Comment:
If you had more than a couple hundred [ in a row, signed integer overflow would occur. How concerned do we need to be about that?

Reply:
We could support that up to a reasonable limit. Perhaps before incrementing, check if overflow would occur (equals to INT8_MAX) and if so return an error. I haven't checked what would happen if signed overflow occurred but I presume it's not great.

Reply:
Then I should probably just stop it at 4 and error at that, since even that depth is enough to be invalid.

Reply:
There. Let me know if the latest change is satisfactory @isaacbrodsky